### PR TITLE
Changes per meeting with Ops team - Closes #624

### LINKED ISSUE #552: Ask Ops team for assistance with incubator deployment
Dependencies:
#550 
#551

### LINKED ISSUE #624: VRMS v0.3 Hosting Requests from Ops team

### PR DESCRIPTION
**Pull request by jasonwong26**
# Summary
Per request from @alex-anakin, deleted and recreated prior PR.

I did my best to address all concerns raised in that PR.  And with the GitHub environment changes, hopefully this will be squeaky clean!

# Notes from prior PR

Met with Darren and the Ops team, and they asked for a few changes so they can move forward with hosting v0.3:

- Provide them with all necessary ENVIRONMENT variables (DONE)
- Convert settings for `MailHog` to ENVIRONMENT variables
- Set up new Production specific Docker files (`backend/Dockerfile.prod` and `client.Dockerfile.prod`)
- Create a set of health check pages (`/healthcheck` and `api/healthcheck`)

## New Environment Variables for MailHog
These changes require the following be added to the backend .env file:

```
MAILHOG_PORT=1025
MAILHOG_USER=user
MAILHOG_PASSWORD=password
```

We'll also want to update the Environment variables document once these changes are approved.

## New Docker Files for Dev

- `backend/Dockerfile.dev`
- `client/Dockerfile.dev`

Darren recommended that we simplify the Dev environment with these (to replace the original `.api` and `.client` files), however I got `nginx` errors when I tested them.
I kept them in this PR for reference, but we'll need additional changes to convert over.

### CODE REVIEW

**Comment by jasonwong26** [comment]
Huzzah!  Success.

**Comment by drubgrubby** [comment]
@jasonwong26 =>  Apologies for not reviewing this yet.  I meant to look at it today, but the day had a mind of its own.  I will review after the meeting tonight.

File: backend/app.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const morgan = require('morgan');
 const cookieParser = require("cookie-parser");
 
 const customRequestHeaderName = 'x-customrequired-header';
-const dontCheckCustomRequestHeaderApis = ["GET::/api/recurringevents"];
+const dontCheckCustomRequestHeaderApis = ["GET::/api/recurringevents", "GET::/api/healthcheck"];
```
Comment (drubgrubby) [inline backend/app.js:12]: Given our recent experience getting `api/recurringevents` to work, should this be `/api/recurringEvents`?  That this worked on your machine and not mine makes me think it's probably fine as is, but I'm just putting it out there.

**Review by drubgrubby — APPROVED** [review]
@jasonwong26 -> This all looks good to me.  

- I made the one note about the recurringevents/recurringEvents question.
- I notice that I can't sign in with 0.4, but I can't sign in there on the current development branch either, so I think that's just the way things are at the moment and nothing to do with your changes. 

I'm going to approve this as it seems good to me, but Alex should definitely review before merging.

**Comment by jasonwong26** [comment]
Thanks @drubgrubby!

I can confirm that `/api/recurringevents` works on my environment.

I'm looking at the signin issue for v0.4 - that was one of the issues I mentioned in the call last night that kept me from making serious progress on writing tickets.

@alex-anakin 

If you can take a look at this and give your approval I'd love to get this merged so we an follow up with the Ops team.

File: backend/app.js (hunk 1 above)
Comment (alex-anakin) [reply backend/app.js:12]: @drubgrubby it's about the internal checking function so we can ignore the case of the letter here.

**Review by alex-anakin — APPROVED** [review]
Sorry for the delay with the review. Everything looks good to me.

**Pull request merged**